### ISSUE TITLE
Issue with "JavaScript basics": final code example doesn't show correct name initially

### ISSUE THREAD

**Issue by EverestRain** [mdn/content#14812]
MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web/JavaScript_basics

#### What information was incorrect, unhelpful, or incomplete?
Not necessarily an error. The final piece of code is supposed to re-prompt the client if a name isn't entered. If you don't enter a name and just press "OK" or "cancel", the prompt will show again. Once you enter a name it should show that name in h1. Instead, it will show "null" or be blank, depending on the button pressed, until you change the name again.

#### Specific section or headline?
https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web/JavaScript_basics#a_user_name_of_null

#### What did you expect to see?
I don't know if having it just display the entered name on the first time after choosing to leave it blank would be too much code for a beginner, but it did scare me that I had typed the code incorrectly until I tried it again. Maybe if nothing else, just put a note that that will happen.

#### Did you test this? If so, how?
I've been using Microsoft Edge and Chrome to display my test page, and it happens on both of them.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/learn/getting_started_with_the_web/javascript_basics`
* MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web/JavaScript_basics
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/learn/getting_started_with_the_web/javascript_basics/index.md
* Last commit: https://github.com/mdn/content/commit/a745cf8d49e6db1bdc283aa08b6dadd11867769a
* Document last modified: 2022-02-02T01:54:36.000Z

</details>

**Comment by wbamberg** [comment]
> The final piece of code is supposed to re-prompt the client if a name isn't entered. If you don't enter a name and just press "OK" or "cancel", the prompt will show again. Once you enter a name it should show that name in h1. Instead, it will show "null" or be blank, depending on the button pressed, until you change the name again.

Sorry, I'm not sure I understand. Here's what I see:

* I open https://mdn.github.io/beginner-html-site-scripted/, and it asks me for my name
* if I click `OK` without entering a name, or `Cancel`, it asks me again
* when I enter a name and click `OK`, it shows the greeting with my name.

This seems correct. Is this not what you are seeing?

(It would be less annoying if it displayed some fallback when you click `Cancel`, since IMO `Cancel` means "stop bothering me", but that's another question...)

**Comment by EverestRain** [comment]
I tried your link through chrome on my phone and it worked fine.
On my computer I would click ok without typing, it would reprompt, and I
would enter a name. When I did that on my computer, it would display
"Mozilla is cool,". The same for cancel, but with null. I would need to get
the prompt to display again and re-enter something for it to display my
entry.
This happened on both chrome and edge. I don't know if there is some
default browser setting that would cause that.
